node/pkg/fetcher/fetcher.go

```diff
@@ -253,6 +253,45 @@ func (f *Fetcher) fetch(adapter AdapterDetail) ([]float64, error) {
 	return data, nil
 }
 
+func (f *Fetcher) requestFeed(definition Definition) (interface{}, error) {
+	if len(f.Proxies) > 0 {
+		return f.requestWithProxy(definition)
+	}
+	return f.requestWithoutProxy(definition)
+}
+
+func (f *Fetcher) requestWithoutProxy(definition Definition) (interface{}, error) {
+	return utils.GetRequest[interface{}](definition.Url, nil, definition.Headers)
+}
+
+func (f *Fetcher) requestWithProxy(definition Definition) (interface{}, error) {
+	var proxies []Proxy
+	if definition.Location != nil && *definition.Location != "" {
+		proxies = f.filterProxyByLocation(*definition.Location)
+	} else {
+		proxies = f.Proxies
+	}
+	if len(proxies) < 1 {
```

Comment:
I think it;s not a good practice to `requestWhithoutProxy` inside of `requestWithProxy`
How about moving conditions to `requestFeed`

Reply:
yes, I think that'd better 👍